src/gui/MainWindow.cpp

```diff
@@ -994,9 +992,57 @@ void MainWindow::hideYubiKeyPopup()
     setEnabled(true);
 }
 
+void MainWindow::bringToFront()
+{
+    ensurePolished();
+    setWindowState(windowState() & ~Qt::WindowMinimized);
+    show();
+    raise();
+    activateWindow();
+}
+
 void MainWindow::handleScreenLock()
 {
     if (config()->get("security/lockdatabasescreenlock").toBool()){
         lockDatabasesAfterInactivity();
     }
 }
+
+QStringList MainWindow::kdbxFilesFromUrls(const QList<QUrl>& urls)
+{
+    QStringList kdbxFiles;
+    for (const QUrl& url: urls) {
+        const QFileInfo fInfo(url.toLocalFile());
+        const bool isKdbxFile = fInfo.isFile() && fInfo.suffix().toLower() == "kdbx";
+        if (isKdbxFile) {
+            kdbxFiles.append(fInfo.absoluteFilePath());
+        }
+    }
+
+    return kdbxFiles;
+}
+
+void MainWindow::dragEnterEvent(QDragEnterEvent* event)
+{
+    const QMimeData* mimeData = event->mimeData();
+    if (mimeData->hasUrls()) {
+        const QStringList kdbxFiles = kdbxFilesFromUrls(mimeData->urls());
+        if (!kdbxFiles.isEmpty()) {
+            event->acceptProposedAction();
+        }
+    }
+}
+
+void MainWindow::dropEvent(QDropEvent* event)
+{
+    const QMimeData* mimeData = event->mimeData();
+    if (mimeData->hasUrls()) {
+        const QStringList kdbxFiles = kdbxFilesFromUrls(mimeData->urls());
+        if (!kdbxFiles.isEmpty()) {
+            event->acceptProposedAction();
+        }
+        for(const QString &kdbxFile: kdbxFiles) {
```

Comment:
kdbxFiles inside kdbxFilesFromUrls() has gone out of scope here, so reference count should be 1 again, but use asConst() around kdbxFiles nevertheless. Also add a space after for.

Reply:
@phoerious  It would be nice if someone of the core developers have added information about how to use range-based for loop with Qt containers to the CONTRIBUTING.md

Good article about [Qt containers and C++11 range-based loops](https://www.dvratil.cz/2015/06/qt-containers-and-c11-range-based-loops/)

> Also add a space after for and move the & sign to the type name.

done

Reply:
Perhaps we should do that. You need to some understanding of how reference counters and detaching work in Qt. It's not KeePassXC stuff though. It's general Qt knowledge.

Reply:
Of course these knowledge relates to Qt. But I'm sure that not everyone who sends patches understands how Qt lists and range-based for loops works :(

Reply:
That's what code reviews are for. :-)